internal/jimm/bootstrap/bootstrap_job_test.go

```diff
@@ -0,0 +1,796 @@
+// Copyright 2025 Canonical.
+package bootstrap_test
+
+import (
+	"time"
+
+	qt "github.com/frankban/quicktest"
+	"github.com/google/uuid"
+	jujucloud "github.com/juju/juju/cloud"
+	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/jujuclient"
+	jujuparams "github.com/juju/juju/rpc/params"
+	"go.uber.org/mock/gomock"
+
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/jimm/bootstrap"
+	"github.com/canonical/jimm/v3/internal/jujuclistore"
+	"github.com/canonical/jimm/v3/internal/jujucommands"
+)
+
+var (
+	jobParams = bootstrap.JobParams{
+		JujuDataDir:          "/path/to/a/juju/data/dir",
+		CLIVersion:           "3.6.9",
+		CLIOs:                "linux",
+		CLIArch:              "aarch64",
+		CloudNameAndRegion:   "special-cloud",
+		ControllerName:       "a",
+		AgentVersion:         "3.6.3",
+		BootstrapTimeout:     0,
+		CloudCred:            jujucloud.CloudCredential{},
+		PersonalCloud:        jujucloud.Cloud{},
+		LoginTokenRefreshURL: "jimm.com/.well-known/jwks.json",
+	}
+)
+
+func pollJob(c *qt.C, s *bootstrapManagerSuite, id uuid.UUID, expectedStatus dbmodel.JobStatus) {
+	var status dbmodel.JobStatus
+	var pollerr error
+	for i := 0; i < 20; i++ {
+		status, pollerr = s.db.GetJobStatus(c.Context(), id)
+		c.Assert(pollerr, qt.IsNil)
+		if status == expectedStatus {
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+	c.Assert(status, qt.Equals, expectedStatus)
+}
+
+func assertJobError(c *qt.C, s *bootstrapManagerSuite, id uuid.UUID, errStr string) {
+	entry := &dbmodel.JobTrackerEntry{JobID: id}
+	err := s.db.GetJob(c.Context(), entry)
+	c.Assert(err, qt.IsNil)
+	c.Assert(entry.Error, qt.Equals, errStr)
+}
+
+func (s *bootstrapManagerSuite) TestBootstrapJob(c *qt.C) {
+	testCtx := c.Context()
+
+	binaryPath := "/faketmp/juju"
+	testOutputLine := "test-line"
+
+	binary := &jujuclistore.Binary{FullPath: binaryPath}
+	binary.RefCount().Store(1) // Simulate the Get().
```

Comment:
I don't think you should be exposing this detail just to test it. I assume you're doing this to verify that you're calling cli.Done()? Can we mock that instead?